src/pages/guides/develop/use_cases.md

```diff
@@ -100,47 +100,75 @@ document.getElementById("anchor").href = downloadUrl;
 
 #### Example:
 
 ```js
+import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
+const { app, constants } = addOnUISdk;
+const { ButtonType, Range, RenditionFormat, RenditionType, RenditionIntent } = constants;
+
 const showPremiumContentError = async () => {
-  const { ButtonType } = addOnUISdk.constants;
-  const {buttonType} = await window.addOnUISdk.app.showModalDialog({
+  const { buttonType } = await window.addOnUISdk.app.showModalDialog({
     variant: "error",
     title: "Export failed",
     description: "Sorry, we were not able to export your design. Some assets are only included in the Premium plan. Try replacing with something else or upgrading Adobe Express to a Premium plan.", 
     buttonLabels: { secondary: "Upgrade" }
   });
   if (buttonType === ButtonType.cancel) return;
   if (buttonType === ButtonType.secondary) {
-    window.open("https://www.adobe.com/go/express_addons_pricing", "_blank")
+    // Deprecated method
+    // ❌ window.open("https://www.adobe.com/go/express_addons_pricing", "_blank")
+    // 👇 Use startPremiumUpgradeIfFreeUser() instead 
+    const hasUpgradedToPremiumUser = await app.startPremiumUpgradeIfFreeUser();
+    if (!hasUpgradedToPremiumUser) {
+      // User did not upgrade, show error dialog
+    } else {
+      // User has upgraded, proceed with export
+      const renditionOptions = {range: Range.entireDocument, format: RenditionFormat.png};
+      renditions.forEach(rendition => { /* do your thing w/ the renditions */ });
+    }
   }
 }
 
 document.querySelector("#export").onclick = async () => {
-  const { app, constants } = addOnUISdk;
-  const { Range, RenditionFormat, RenditionType, RenditionIntent } = constants;
-  /* THE FOLLOWING FLAG CAN BE USED FOR TESTING PURPOSES ONLY -- REMOVE BEFORE RELEASE */
-  app.devFlags.simulateFreeUser = true; 
-  const renditionOptions = {range: Range.currentPage, format: RenditionFormat.png};
-  try {
-    const renditions = await app.document.createRenditions(renditionOptions);
-    renditions.forEach(rendition => { /* do your thing w/ the renditions */ });
-  } catch (err) {
-    if (err.message?.includes("USER_NOT_ENTITLED_TO_PREMIUM_CONTENT")) {
-      showPremiumContentError();
+  // 👇 Testing purposes only!
+  app.devFlags.simulateFreeUser = true; // Remove this line in production!
+  
+  const userIsPremium = await app.currentUser.isPremiumUser();
+  const pages = await app.document.getPagesMetadata({
+    range: Range.entireDocument, // use the proper range
+  });
+  let containsPremiumContent = false;
+  for (const page of pages) {
+    if (page.hasPremiumContent) {
+      containsPremiumContent = true;
+      break;
     }
   }
+
+  if (containsPremiumContent && !userIsPremium) {
+    showPremiumContentError();
+    return;
+  } else {
+    const renditionOptions = {range: Range.entireDocument, format: RenditionFormat.png};
```

Comment:
This pattern requires the dev to have the export code in two places. We also lose the call to `createRenditions` which is is still required (before we can iterate over the `renditions` array).

I think it'd be good to pull out the export code into a separate function (that way error handling can be located in a single place)

I also just realized that if the user is in a co-editing scenario, they could still pass all the premium checks only to fail at the last minute when trying to create renditions, so we probably still need the try/catch section.

I think the following code would handle all the possible failure modes (but please double check me!)
```
import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
const { app, constants } = addOnUISdk;
const { ButtonType, Range, RenditionFormat, RenditionType, RenditionIntent } = constants;

const showPremiumContentError = async () => {
  const { buttonType } = await window.addOnUISdk.app.showModalDialog({
    variant: "error",
    title: "Export failed",
    description: "Sorry, we were not able to export your design. Some assets are only included in the Premium plan. Try replacing with something else or upgrading Adobe Express to a Premium plan.", 
    buttonLabels: { secondary: "Upgrade" }
  });

  if (buttonType === ButtonType.cancel) return false; // user is still not premium
  if (buttonType === ButtonType.secondary) {
    // Original flow (don't use anymore)
    // ❌ window.open("https://www.adobe.com/go/express_addons_pricing", "_blank")
    // 👇 Use startPremiumUpgradeIfFreeUser() instead 
    const hasUpgradedToPremium = await app.startPremiumUpgradeIfFreeUser();
	return hasUpgradedToPremium;
  }
}

const isRangeSafeToExport = await (range) => {
  const userIsPremium = await app.currentUser.isPremiumUser();
  const pages = await app.document.getPagesMetadata({range});
  const containsPremiumContent = pages.some(page => page.hasPremiumContent);
  return (containsPremiumContent && userIsPremium) || !containsPremiumContent;  
}

const exportDocument = async () => {
  // 👇 Testing purposes only!
  app.devFlags.simulateFreeUser = true; // Remove this line in production!

  let isSafeToExport = isRangeSafeToExport(Range.entireDocument);  
  if (!isSafeToExport) {
    const isNowPremiumUser = await showPremiumContentError();
    isSafeToExport = isNowPremiumUser;
  }
  
  if (isSafeToExport) {
    try {
      const renditions = await app.document.createRenditions({
        range: Range.entireDocument, format: RenditionFormat.png
      });
      renditions.forEach(rendition => { /* do your thing w/ the renditions */ });	  
    } catch (err) {
      // did someone just add premium content in the split second between
      // our original check? did the user just downgrade?
      if (err.message?.includes("USER_NOT_ENTITLED_TO_PREMIUM_CONTENT")) {
        return await exportDocument(); // try again
      }
    }
  }  
}

document.querySelector("#export").onclick = exportDocument;
```

Reply:
Oh, I see what you're doing. It's very neat!!
I've fixed a couple of typos (an `await` that was meant to be `async` and a missing `await` when invoking `isRangeSafeToExport()`); other than that, it is perfect, thanks!